http-agent/src/main/java/org/motechproject/http/agent/listener/HttpClientEventListener.java

```diff
@@ -200,6 +200,13 @@ public void handle(MotechEvent motechEvent) {
         try {
             retValue = r.call();
         } catch (Exception e) { // Http request failed for all retries
+            if (e instanceof HttpException && !(e.getCause().toString().contains("I/O error"))) {
+                String cause = e.getCause().toString();
+                cause = cause.substring(cause.indexOf(":") + 2, cause.indexOf(":") + 5);
```

Comment:
Crafting the cause like this is super unintuitive. Moreover we cannot predict all exception causes, so this can produce anything.